scipy/stats/tests/test_distributions.py

```diff
@@ -7214,6 +7214,34 @@ def test_delta_cdf(self, scale, expected):
         assert_allclose(delta, expected, rtol=1e-13)
 
 
+    @pytest.mark.parametrize('n', range(6))
+    @pytest.mark.parametrize('c', [1.6, 2.6])
+    @pytest.mark.parametrize('d', [0.1, 1, 2])
+    def test_moments(self, n, c, d):
+        # gh-18838 reported that burr12 moments could be invalid. This was
+        # because moment n exists iff shape parameters c*d > n, but this was
+        # not checked in the implementation. Confirm that this is resolved.
+        res = stats.burr12(c, d).moment(n)
+        if c*d > n:
+            assert np.isfinite(res)
+        else:
+            assert np.isnan(res)
```

Comment:
See e,g, [here](https://www.jstor.org/stable/2335782?seq=2) page 4:
![image](https://github.com/scipy/scipy/assets/6570539/8db6ff74-2f6b-495c-9b5f-65eaad64745a)

and [here](![image](https://github.com/scipy/scipy/assets/6570539/6bcb0235-3bd4-4f9c-b4cc-044cb2f23baa)) page 27:
![image](https://github.com/scipy/scipy/assets/6570539/72cac315-4436-4dfa-994f-6631f32cca2e)